theano/gof/op.py

```diff
@@ -780,6 +780,10 @@ class Op(utils.object2, PureOp, CLinkerOp):
 
     """
     def __new__(cls, *args, **kwargs):
+        # TODO: Ensuring all the Ops created henceforth has __props__
+        #       This helps to ease the process of caching by making use
+        #       of the _props_dict() instead of inferring the parameters
+        #       passed to the class's constructor.
```

Comment:
I wanted this earlier mainly thinking of the Ops inside elemwise so that it could be cached. Now we're using the frozen dict approach and this isn't needed anymore